src/string_pool.cr

```diff
@@ -1,19 +1,52 @@
+# A String pool is a collection of Strings
+# It can be used to return an already present String instead of creating a new String.
+#
+#
+#   require "string_pool"
+#   stringpool = StringPool.new
```

Comment:
Maybe use `string_pool` or just `pool` for the variable name. Without `_` it feels strange.